git-extra/notepad

```diff
@@ -15,7 +15,14 @@ die "Usage: $0 <file>"
 
 if test -f "$1"
 then
-	unix2dos.exe "$1"
+	commitencoding="$(git config --get i18n.commitencoding)"
+	if [ $? -ne 0 ]  ||
+		[ "$commitencoding" = "utf-8" ] ||
+		[ "$commitencoding" = "utf8" ]; then
+		unix2dos.exe -m "$1"
+	else
+		unix2dos.exe "$1"
+	fi
```

Comment:
Isn't this wrapper also used for other tasks that might be UTF encoded? How well does `unix2dos` handle files that already are in UTF-8BOM with the `-m` flag? Could Notepad also mess up with other UTF encodings? How well does this solution handle edge cases like unexpected crashes? Would something like this, but modified to accomodate more situations than just commit messages be a better solution?

````diff
then
	unix2dos.exe "$1"
fi &&
-notepad.exe "$1" &&
+case "$(git config --get i18n.commitencoding)" in
+utf*)
+		notepad.exe /W "$1"
+*)
+		notepad.exe "$1"
+esac &&
 dos2unix.exe "$1" &&
 case "$1" in
 */COMMIT_EDITMSG|*\\COMMIT_EDITMSG)
````

Sources for the `/W` argument:
[1](http://answers.microsoft.com/en-us/windows/forum/windows_7-files/notepadexe-command-line-options/810760c1-a45a-4013-9544-1c1208e1b389)
[2](http://alinconstantin.blogspot.de/2012/08/notepadexe-command-line-arguments.html)
[3](http://superuser.com/questions/131826/notepad-switches)